pkgs/development/compilers/ldc/default.nix

```diff
@@ -0,0 +1,107 @@
+{ stdenv, fetchFromGitHub, cmake, llvm, dmd, curl, tzdata, python,
+  lit, gdb, unzip, darwin }:
+
+stdenv.mkDerivation rec {
+  name = "ldc-${version}";
+  version = "1.3.0";
+
+  srcs = [
+  (fetchFromGitHub {
+    owner = "ldc-developers";
+    repo = "ldc";
+    rev = "v${version}";
+    sha256 = "1ac3j4cwwgjpayhijxx4d6478bc3iqksjxkd7xp7byx7k8w1ppdl";
+  })
+  (fetchFromGitHub {
+    owner = "ldc-developers";
+    repo = "druntime";
+    rev = "ldc-v${version}";
+    sha256 = "1m13370wnj3sizqk3sdpzi9am5d24srf27d613qblhqa9n8vwz30";
+  })
+  (fetchFromGitHub {
+    owner = "ldc-developers";
+    repo = "phobos";
+    rev = "ldc-v${version}";
+    sha256 = "0fhcdfi7a00plwj27ysfyv783nhk0kspq7hawf6vbsl3s1nyvn8g";
+  })
+  (fetchFromGitHub {
+    owner = "ldc-developers";
+    repo = "dmd-testsuite";
+    rev = "ldc-v${version}";
+    sha256 = "0dmdkp220gqhxjrmrjfkf0vsvylwfaj70hswavq4q3v4dg17pzmj";
+  })
+  ];
+
+  sourceRoot = ".";
+
+  postUnpack = ''
+      mv ldc-v${version}-src/* .
+
+      mv druntime-ldc-v${version}-src/* runtime/druntime
+
+      mv phobos-ldc-v${version}-src/* runtime/phobos
+
+      mv dmd-testsuite-ldc-v${version}-src/* tests/d2/dmd-testsuite
+
+      # Remove cppa test for now because it doesn't work.
+      rm tests/d2/dmd-testsuite/runnable/cppa.d
+      rm tests/d2/dmd-testsuite/runnable/extra-files/cppb.cpp
+  '';
+
+  postPatch = ''
+      substituteInPlace runtime/phobos/std/net/curl.d \
+          --replace libcurl.so ${curl.out}/lib/libcurl.so
+
+      # Ugly hack to fix the hardcoded path to zoneinfo in the source file.
+      # https://issues.dlang.org/show_bug.cgi?id=15391
+      substituteInPlace runtime/phobos/std/datetime.d \
+          --replace /usr/share/zoneinfo/ ${tzdata}/share/zoneinfo/
+  ''
+
+  + stdenv.lib.optionalString stdenv.hostPlatform.isDarwin ''
+      substituteInPlace driver/tool.cpp \
+          --replace "gcc" "clang"
+  '';
+
+  nativeBuildInputs = [ cmake llvm dmd python lit gdb unzip ]
+
+  ++ stdenv.lib.optional stdenv.hostPlatform.isDarwin (with darwin.apple_sdk.frameworks; [
+    Foundation
+  ]);
+
+  buildInputs = [ curl tzdata stdenv.cc ];
+
+
+  configurePhase = ''
+    mkdir build && cd build
+    cmake \
+    -DCMAKE_INSTALL_PREFIX=$out \
+    -DINCLUDE_INSTALL_DIR=$out/include/dlang/ldc \
+    ..
+  '';
+
+  buildPhase = ''
+    export DMD=$PWD/bin/ldc2
+    make -j$NIX_BUILD_CORES DMD=$DMD
+  '';
+
+  doCheck = true;
+
+  checkPhase = ''
+      export DMD=$PWD/bin/ldc2
+      # Exclude dmd-testsuite for now because it has problems on Linux and Darwin
+      ctest -j $NIX_BUILD_CORES -V DMD=$DMD
+  '';
+
+  installPhase = ''
+    make install
+  '';
+
+  meta = with stdenv.lib; {
+    description = "The LLVM-based D compiler";
+    homepage = https://github.com/ldc-developers/ldc;
+    # from https://github.com/ldc-developers/ldc/blob/master/LICENSE
+    license = with licenses; [ bsd3 boost mit ncsa gpl2Plus ];
```

Comment:
Do you want to maintain it?

Reply:
I thought about that.
Not sure what it means in reality though.
In fact I plan to keep all D related development stuff going as time permits.
I don't like it either that there is no maintainer for all D related stuff.
So i guess the difference is just a formality right? (I know about adding my name into maintainers list, I already did it in a PR a few year back which was never merged fully)

Reply:
You might be consulted in pull requests as a reviewer and you get emails from hydra, when the build of this compiler fails. How much time you want to spent is up to you. Beeing a package maintainer in nixpkgs has usually less implications compared to other linux/package distributions.

Reply:
Sounds nice.
I added myself to the maintainers. Thanks.